DOCSP-20678 General connection page (#379)
mongodb/docs-compass: source/connect/general-connection.txt

   :backlinks: none
   :depth: 1
   :class: singlecol

The :guilabel:`General` connection tab allows you to select a 
:guilabel:`Connection String Scheme` for your |compass-short| deployment. 

> Connecting

Reviewer: Suggest making this "Procedure".

Author: Done.

   :backlinks: none
   :depth: 1
   :class: singlecol

The :guilabel:`General` connection tab allows you to select a 

> :guilabel:`Connection String Scheme` for your |compass-short| deployment. 

Reviewer 2: I think we might be able to be more specific about how the configurations are used to connect to MongoDB, how's something like this?
```suggestion
:guilabel:`Connection String Scheme` and configure the hostname or hosts used to connect to your MongoDB deployment. 
```

Author: Done.

Procedure
---------

.. procedure::
   :style: normal

   .. step:: Click :guilabel:`Advanced Connection Options`.

      .. figure:: /images/compass/advanced-connection-options.png
         :figwidth: 690px
         :alt: New Advanced Connection Options

   .. step:: Click the :guilabel:`General` tab.

      .. procedure::
         :style: connected

         .. step:: Select a Connection String Scheme.

            You can select one of the following options:

            .. list-table::
               :header-rows: 1
               :class: border-table

               * - Connection String Scheme
                 - Definition

               * - mongodb
                 - :manual:`Standard Connection String Format </reference/connection-string/>`. 
                   The standard format of the MongoDB connection URI is used to 
                   connect to a MongoDB deployment: standalone, replica set, or a sharded 

>                    cluster. 

Reviewer 3: This information might be a bit confusing, because the +srv variant is also used for the exact same things.

Maybe this could be rephrased as “The standard format of the MongoDB connection URI is used to connect to a MongoDB deployment *by specifying its hosts directly*”.

Author: Done.

               * - mongodb+srv 
                 - :manual:`DNS Seed List Connection Format 
                   </reference/connection-string/#dns-seed-list-connection-format>`. 
                   The ``+srv`` indicates to |compass| that the hostname that follows 
                   corresponds to a DNS SRV record.

         .. step:: Enter a Hostname.

>             In the :guilabel:`Host` field, enter the hostname of the server where the 

Reviewer 2: The host input allows a user to specify multiple hosts if they're not connecting with an srv connection. Maybe we can include something like this to describe that:
```suggestion
            In the :guilabel:`Host` field, enter the hosts or hostname of the server where the 
```

Author: Done.

            deployment is running.

            If you are running your deployment locally, this value is ``localhost``. 
            If you are connecting to an Atlas cluster, see 
            :atlas:`Connect to Database Deployment </connect-to-database-deployment/>` 

>             for Atlas-specific guidance on how to connect to |compass-short|.

Reviewer 2: ```suggestion
            for Atlas-specific guidance on how to connect with |compass-short|.
```

            If you are not sure of your hostname, contact your Database 
            Administrator for information.

>          .. step:: Enable Direct Connection. 

Reviewer 3: Should this heading also include some indication that it is optional, like the advanced connection options step below?

Author: Yes, I agree!

            When you use the ``mongodb`` Connection String Scheme, you have the 
            option to enable a :driver:`Direct Connection 
            </node/current/fundamentals/connection/#direct-connection>`. When you 
            enable this setting, |compass-short| runs all operations on the specified 
            host.

            .. important::
               When you specify ``directConnection`` and connect to a secondary member 
               of a replica set, your write operations fail because it is not the 
               primary member.

   .. step:: (Optional) For advanced connection configuration options, click the :ref:`Advanced <advanced-connection-tab>` tab.

   .. step:: Click Connect.

.. seealso::

>    To disconnect your deployment, see :ref:`<disconnect-tab>`. 

Author: Done.

   :backlinks: none
   :depth: 1
   :class: singlecol

The :guilabel:`General` connection tab allows you to select a 
:guilabel:`Connection String Scheme` for your |compass-short| deployment. 

Procedure
---------

.. procedure::
   :style: normal

   .. step:: Click :guilabel:`Advanced Connection Options`.

      .. figure:: /images/compass/advanced-connection-options.png
         :figwidth: 690px
         :alt: New Advanced Connection Options

   .. step:: Click the :guilabel:`General` tab.

      .. procedure::
         :style: connected

         .. step:: Select a Connection String Scheme.

            You can select one of the following options:

            .. list-table::
               :header-rows: 1
               :class: border-table

               * - Connection String Scheme
                 - Definition

               * - mongodb
                 - :manual:`Standard Connection String Format </reference/connection-string/>`. 
                   The standard format of the MongoDB connection URI is used to 
                   connect to a MongoDB deployment: standalone, replica set, or a sharded 
                   cluster. 

               * - mongodb+srv 

>                  - :manual:`DNS Seed List Connection Format </reference/>`. The ``+srv`` 

Reviewer: I'm a little confused why we're linking to the reference page here? Can we instead link to the DNS Seed List section of the Connection String page?

```
:manual:`DNS Seed List Connection Format </reference/connection-string/#dns-seed-list-connection-format>`
```

Author: Done.

                   indicates to |compass| that the hostname that follows corresponds to a 

>                    :manual:`DNS SRV record </reference/connection-string/>`.

Reviewer: If we're providing a link in the initial part of this definition, we probably don't need a second link at the end.

Author: Done.

         .. step:: Enter a Hostname.

            In the :guilabel:`Host` field, enter the hostname of the server where the 
            deployment is running.

            If you are running your deployment locally, this value is ``localhost``. 
            If you are connecting to an Atlas cluster, you can get your hostname 
            from your :atlas:`database deployment detail view 

>             </tutorial/view-all-database-deployments/#view-database-deployment-details>` 

Reviewer: I think this page might be a better place to link: https://docs.atlas.mongodb.com/connect-to-database-deployment/

Author: Done.

            in Atlas.

            If you are not sure of your hostname, contact your Database 
            Administrator for information.

         .. step:: Enable Direct Connection. 

            When you use the ``mongodb`` Connection String Scheme, you have the 
            option to enable a :driver:`Direct Connection 
            </node/current/fundamentals/connection/#direct-connection>`. When you 
            enable this setting, Compass runs all operations on the specified host.

            When you specify ``directConnection`` and connect to a secondary member 
            of a replica set, your write operations fail because it is not the 

>             primary member.

Reviewer: Optional, but I might consider putting this blurb in an `important` admonition.

Author: Done.

               :header-rows: 1
               :class: border-table

               * - Connection String Scheme
                 - Definition

               * - mongodb
                 - :manual:`Standard Connection String Format </reference/connection-string/>`. 
                   The standard format of the MongoDB connection URI is used to 
                   connect to a MongoDB deployment: standalone, replica set, or a sharded 
                   cluster. 

               * - mongodb+srv 
                 - :manual:`DNS Seed List Connection Format 
                   </reference/connection-string/#dns-seed-list-connection-format>`. 
                   The ``+srv`` indicates to |compass| that the hostname that follows 
                   corresponds to a DNS SRV record.

         .. step:: Enter a Hostname.

            In the :guilabel:`Host` field, enter the hostname of the server where the 
            deployment is running.

            If you are running your deployment locally, this value is ``localhost``. 
            If you are connecting to an Atlas cluster, see 
            :atlas:`Connect to Database Deployment </connect-to-database-deployment/>` 

>             for Atlas specific guidance on how to connect to |compass-short|.

Reviewer: [nit]

```suggestion
            for Atlas-specific guidance on how to connect to |compass-short|.
```

Author: Done.

            If you are not sure of your hostname, contact your Database 
            Administrator for information.

         .. step:: Enable Direct Connection. 

            When you use the ``mongodb`` Connection String Scheme, you have the 
            option to enable a :driver:`Direct Connection 
            </node/current/fundamentals/connection/#direct-connection>`. When you 
            enable this setting, |compass-short| runs all operations on the specified 
            host.

            .. important::
               When you specify ``directConnection`` and connect to a secondary member 
               of a replica set, your write operations fail because it is not the 
               primary member.

   .. step:: For more configuration options, see the 
      :ref:`Advanced <advanced-connection-tab>` tab.

      You can specify additional connection configurations under the 

>       :ref:`Advanced <advanced-connection-tab>` tab.

Reviewer: Maybe we can consolidate these lines into something like:

"(Optional) For advanced connection configuration options, click the :ref:`Advanced <advanced-connection-tab>` tab."

Author: Done.
